vispy/visuals/bar.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+from .mesh import MeshVisual
+
+
+class BarVisual(MeshVisual):
+    """Visual that calculates and displays a bar graph
+
+    Parameters
+    ----------
+    data : array-like
+        1st column is the height, optional 2nd column is bottom
+    height : array-like
+        Height of the bars
+    bottom : array-like
+        Bottom of the bars
+    width : int | float
+        Width of all bars
+    shift : int | float
+        Shift of all bars along the x-axis
+    color : instance of Color
+        Color of the bar.
+    orientation : {'h', 'v'}
+        Orientation of the histogram - 'h' is default
+    """
+
+    def __init__(self, height, bottom=None, width=0.8, shift=0.0, color='w', orientation='h'):
+        if bottom is None:
+            bottom = np.zeros(height.shape[0])
+        
+        rr, tris = calc_vertices(height, bottom, width, shift, orientation)
+
+        MeshVisual.__init__(self, rr, tris, color=color)
+
+    def update_data(self, height, bottom=None, width=0.8, shift=0.0, color='w'):
+        if bottom is None:
+            bottom = np.zeros(height.shape[0])
+        rr, tris = calc_vertices(height, bottom, width, shift=shift)
+
+        MeshVisual.set_data(self, rr, tris, color=color)
+
+
+def calc_vertices(height, bottom, width, shift, orientation):
+
+    y_position = np.zeros((height.shape[0], 2))
+    
+    y_position[:, 0] = height
+    y_position[:, 1] = bottom
+
+    y_position = y_position.flatten().repeat(2)
+
+    stack_n_x2 = np.arange(0, height.shape[0], dtype=int) 
+    stack_n_x2 = np.expand_dims(stack_n_x2, axis=1)
+    stack_n_x2 = np.repeat(stack_n_x2, 1, axis=1)
+    stack_n_x2 = np.repeat(stack_n_x2, 2, axis=0)
```

Comment:
If I understand correctly, this was the goal? Maybe this way is a bit more intuitive.

```suggestion
    stack_n_x2 = np.arange(height.shape[0]).repeat(4).reshape(-1, 1)
```

By setting `repeat(4)` instead of `2` we don't have to repeat again later, which is both faster for memory allocation and clearer in the code.

Reply:
As you saw its used later, however i'd say 
`stack_n_x2 = np.arange(height.shape[0]).repeat(2).reshape(-1, 1)`
would be better
I did a quick test: 

`%timeit old(np.arange(0, 250, 0.5))
%timeit new(np.arange(0, 250, 0.5))
`
12.8 µs ± 1.18 µs per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
7.23 µs ± 689 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

So its about 50% faster. 

if I inclue the second repeat(2) from line 71/75 its twice as fast

25.2 µs ± 5.47 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each) # original commit
12.4 µs ± 1.78 µs per loop (mean ± std. dev. of 7 runs, 100,000 loops each) # repeat(2).repeat(2)
7.69 µs ± 784 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each) # repeat(4)